### PR TITLE
Logstash 5.x compatible

### PR DESCRIPTION
**Pull request by vktor18**
fixed mutex from #11 and added support for Logstash 5.x

### CODE REVIEW

**Comment by wiibaa** [comment]
@vktor18 to properly run the travis build, you must update the travis.yml to match what is done in other plugin https://github.com/logstash-plugins/logstash-filter-grok/blob/master/.travis.yml mainly forcing jdk8 is key

**Comment by vktor18** [comment]
I have no idea why travis is angry with me.... Did I push incorrect changes for the travis.yml?

**Comment by wiibaa** [comment]
Now the travis build is ok, but the spec is failing

**Comment by ph** [comment]
@vktor18 This seems to be a real error.

```
1) LogStash::Filters::Cipher 1000 events, 11 re-use, encrypt/decrypt aes-128-cbc, 16b RANDOM IV, 16b key, b64 encode validate encrypted message is not equal to message
     Failure/Error: expect(result.get("message")).not_to eq(result.get("message_decrypted"))
```

We are expecting the `message` field not to match the encrypted version in the `message_decrypted`

I am wondering if the line at https://github.com/logstash-plugins/logstash-filter-cipher/pull/18/files#diff-8046de15c827a600856f2cdb65db73edR153 is doing what we are actually expecting.

**Comment by vktor18** [comment]
thanks @ph for the reply. 

I don't understand which line are you referring in the link ..

**Comment by ph** [comment]
@vktor18 

I am referring to theses lines, I wonder if they are problematic with the LS 5.0 API.

```ruby
 if (event.get(@source).nil? || event.get(@source).empty?)
          @logger.debug("Event to filter, event 'source' field: " + @source + " was null(nil) or blank, doing nothing")
          return
        end
```

**Comment by wiibaa** [comment]
@vktor18 would be nice to add `.idea/` to the git ignore

File: lib/logstash/filters/cipher.rb [hunk 1]
```diff
@@ -1,15 +1,41 @@
-# encoding: utf-8
+# encoding: ASCII-8BIT
```
Comment (wiibaa) [inline lib/logstash/filters/cipher.rb:1]: Was it changed by your IDE, or needed in your code for some reason ?

Comment (vktor18) [reply lib/logstash/filters/cipher.rb:1]: I changed because I had an error of incompatible type.

**Pull request closed without merging**